main.go

```diff
@@ -51,6 +64,66 @@ func (c *runCmd) Run() error {
 	return nil
 }
 
+func (c *fmtCmd) Run() error {
+	if len(c.Files) == 0 {
+		return format(os.Stdin, os.Stdout, c.Check)
+	}
+	var out io.StringWriter = os.Stdout
+	for _, filename := range c.Files {
+		in, err := os.Open(filename)
+		if err != nil {
+			return err
+		}
+		if c.Write {
+			out, err = os.CreateTemp("", "evy")
+			if err != nil {
+				return fmt.Errorf("%s: %w", filename, err)
+			}
+		}
+		if err := format(in, out, !c.Write); err != nil {
+			return fmt.Errorf("%s: %w", filename, err)
+		}
+		if err := in.Close(); err != nil {
+			return err
+		}
+		if c.Write {
+			tempFile := out.(*os.File)
+			if err := tempFile.Close(); err != nil {
+				return fmt.Errorf("%s: %w", filename, err)
+			}
+			if err := os.Rename(tempFile.Name(), filename); err != nil {
+				return fmt.Errorf("%s: %w", filename, err)
+			}
+		}
+	}
+	return nil
+}
+
+func format(r io.Reader, w io.StringWriter, exitError bool) error {
```

Comment:
nit: I think I would call `exitError` `checkOnly` - uses the same terminology as the flags